data/creaturescripts/scripts/reward_chest.lua

```diff
@@ -0,0 +1,304 @@
+	local Query = db.storeQuery('SELECT id, reward, date FROM player_rewardchest WHERE player_id = '.. self:getGuid() ..';')
+	if Query ~= false then
+		repeat
+			local rewardBag = unserialize(result.getDataString(Query, 'reward'))
+			self:addReward(rewardBag, result.getDataInt(Query, 'date'), result.getDataInt(Query, 'id'))
+		until not result.next(Query)
+		result.free(Query)
+	end
+	local depotChest = self:getDepotChest(REWARD_CHEST.DEPOT_ID, true)
+	for index = (depotChest:getSize() - 1), 0, -1 do
+	    local container = depotChest:getItem(index)
+	    if (container:getAttribute('text') ~= nil) and (tonumber(container:getAttribute('text')) + REWARD_CHEST.EXPIRE < os.time()) then
+	    	container:remove()
+	    end
+	end
+	return self:setExhaustion(REWARD_CHEST.STORAGE, REWARD_CHEST.EXHAUST)
+end
+
+function string.diff(self)
+    local format = {
+        {'day', self / 60 / 60 / 24},
+        {'hour', self / 60 / 60 % 24},
+        {'minute', self / 60 % 60},
+        {'second', self % 60}
+    }
+
+    local out = {}
+    for k, t in ipairs(format) do
+        local v = math.floor(t[2])
+        if(v > 0) then
+            table.insert(out, (k < #format and (#out > 0 and ', ' or '') or ' and ') .. v .. ' ' .. t[1] .. (v ~= 1 and 's' or ''))
+        end
+    end
+    local ret = table.concat(out)
+    if ret:len() < 16 and ret:find('second') then
+        local a, b = ret:find(' and ')
+        ret = ret:sub(b+1)
+    end
+    return ret
+end
+
+function doBossReward(monster, corpse)
+    REWARD_CHEST.LOOT[corpse:getUniqueId()] = {}
+    corpse:setAttribute('aid', 21584)
+    corpse:setAttribute('text', corpse:getUniqueId())
+    for killer, damage in pairs(monster:getDamageMap()) do
+    	local player, str = Player(killer), 'Loot of ' .. MonsterType(monster:getName()):getNameDescription() .. ': '
+    	if player ~= nil then
+    		local rewardBag = doCreateItemEx(REWARD_CHEST.CONTAINER)
+    		if player:getStamina() > 840 then
+        		local loot = MonsterType(monster:getName()):getBossReward(damage.total/monster:getMaxHealth())
+        		if #loot > 0 then
+        			addContainerItems(Container(rewardBag), loot)
+        			REWARD_CHEST.LOOT[corpse:getUniqueId()][player:getGuid()] = loot
+        			str = str .. Container(rewardBag):getContentDescription()
+        		else
+        			str = str .. 'nothing'
+        		end
+        	else
+        		str = str .. 'nothing (due to low stamina)'
+        	end
+        	player:sendTextMessage(MESSAGE_INFO_DESCR, str .. '.')
+        end
+    end
+    addEvent(doSaveReward, REWARD_CHEST.DECAY[monster:getName():lower()]*1000, corpse:getUniqueId(), MonsterType(monster:getName()):getNameDescription())
+end
+
+function onKill(creature, target)
+    if (Monster(target) ~= nil) and isInArray(REWARD_CHEST.BOSSES, target:getName():lower()) then
+    	local corpse = Item(doCreateItem(MonsterType(target:getName()):getCorpseId(), 1, target:getPosition()))
```

Comment:
doCreateItem is a compat function, so compiling the Item userdata isen't needed.
MonsterType accepts userdata values, so don't fetch the name if you don't need it in other places.